test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests_Java.cs

```diff
@@ -46,6 +46,7 @@ public async Task JavaProcess_Different_AfterHostRestart()
             // Trigger a restart
             await _fixture.Host.RestartAsync(CancellationToken.None);
             await HttpTrigger_Java_Get_Succeeds();
+            Thread.Sleep(TimeSpan.FromSeconds(2));
```

Comment:
why do we need to sleep for 2 seconds?

Reply:
Not sure if the concern is about 2 seconds or sleep. I will share more context here - we were discussing the grace period we want to keep and finalized it to be 10 seconds. Updated the code with it. The test is restarting the service so internally host will wait for 10 seconds for process to get exited and then kill the process. Thus, added the delay.

Reply:
Is there a way to check if the process has exited? Happy for the delay to be the fallback if that is not possible, but generally adding sleeps leads to flakey tests. So if the host has a way of checking if a process has exited, that would be the ideal way to do that

Reply:
Changes in this file not needed now.